common/lib/api-client/index.js

```diff
@@ -24,6 +24,10 @@ module.exports = function() {
     logger: IS_DEV,
   })
 
+  // the endpoint for cancel is structured in a way that requires its own model. However,
+  // devour would pluralise it. This rule ensures that we get /cancel and not /cancels
```

Comment:
I'm wondering whether actually the better solution is to just define the custom path within the model.

That would avoid having to have these comments in two places. I'm just conscious we may have to do this for other status change endpoints, like `/complete` and it might be better to keep those maintained within the models configuration.

So this:

```javascript
  cancel: {
    attributes: {
      timestamp: '',
      notes: '',
      cancellation_reason: '',
      cancellation_reason_comment: '',
    },
  },
```

would become this:

```javascript
  cancel: {
    attributes: {
      timestamp: '',
      notes: '',
      cancellation_reason: '',
      cancellation_reason_comment: '',
    },
    options: {
      collectionPath: 'cancel',
    },
  },
```

What do you think? I can just see us having 5/6/7 of these addUnacountableRules here and it being a bit harder to maintain...

Reply:
Or...

common/lib/api-client/models.js
```js
module.exports = {
  ...
  cancel: {
    attributes: {
      ...
    },
    singular: true
  }
}
```

common/lib/api-client/index.js
```js
const singulars = Object.keys(models).filter(key => models[key].singular)
...
pluralize: (str) => singulars.includes(str) ? str : pluralize(str)
```

Reply:
@solidgoldpig I tried but the injected `pluralize` doesn't have a `.singular` method which is invoked further down the line by the api client. I am sure it can be made working with some hack but doesn't seem like a super convenient solution.

Reply:
@teneightfive seems an improvement in readability, so I have applied your suggestion.